src/main/java/org/kairosdb/core/http/rest/MetricsResource.java

```diff
@@ -438,7 +437,12 @@ public Response get(@Context HttpHeaders httpHeaders, String json) throws Except
 
 						DatastoreQuery dq = datastore.createQuery(query);
 						try {
-							List<DataPointGroup> results = dq.execute();
+							List<DataPointGroup> results;
+							if (query.isRejected()) {
+								results = new ArrayList<>();
+							} else {
+								results = dq.execute();
+							}
 							jsonResponse.formatQuery(results, query.isExcludeTags(), dq.getSampleSize());
```

Comment:
I wonder how `getSampleSize()` is going to behave if we didn't call `execute()`?

Reply:
it's going to stay uninitialized, and the default value of `int` is 0